Encryption.md

```diff
@@ -128,6 +127,26 @@ unique nonce with a length of 12 bytes (96 bits). Notice: the NIST
 specification uses a term “IV” for what is called “nonce” in the Parquet encryption design.
 
 
+#### 4.1.4 Invocation limit
+According to the section 8.3 of the NIST SP 800-38D document, one key can be used for up to 
+2^32 encryption operations with RGB-based nonces in the AES GCM cipher. If this limit is exceeded, 
+the data encryption can be broken. The bulk of modules in a parquet file are page headers and data 
+pages. Therefore, one encryption key must not be applied to more than 2^31 (~2 billion) pages. In 
+parquet files encrypted with multiple keys (footer and column keys), the invocation limit is 
+applied to each key separately.
+
+
+Parquet is a local library, with each instance running in a process unaware of other instances. 
```

Comment:
Parquet is a data format, so this reads a bit strangely. A Parquet writer could be implemented as a centralized service.

Reply:
Agreed.

Reply:
I agree talking about parquet as a library is confusing (perhaps this is conflating the parquet-java implementation and the format)

Perhaps this section could be rephrased as "parquet implementation guidelines"

Perhaps something along the lines of 

"When running in the context of a larger system, any particular parquet writer implementation likely
does not have sufficient context to enforce key invocation limits system-wide. Therefore, 
the higher level system itself must arrange to supply keys appropriately to the various writer instances"